tools/generate_std_names.py

```diff
@@ -62,32 +77,72 @@ def process_name_table(tree, element_name, *child_elements):
     """
     for elem in tree.iterfind(element_name):
         sub_section = {}
-
         for child_elem in child_elements:
-            found_elem = elem.find(child_elem)
-            sub_section[child_elem] = found_elem.text if found_elem is not None else None
-
+            sub_section[child_elem] = found_or_none(elem.find(child_elem))
         yield {elem.get("id") : sub_section}
 
 
-def to_dict(infile, outfile):
-    values = {}
-    aliases = {}
-
+def prettydict(outfile, varname, data):
+    """Pretty formatted output of the data (dict) assigned to the variable 'varname'."""
+    outfile.write(f'{varname} = {{\n')
+    for k, v in dict(sorted(data.items())).items():
+        outfile.write(f'    "{k}": "{v}",\n')
+    outfile.write("}\n\n")
+
+
+def decode_version(outfile, tree):
+    """Decode the version information in the xml header information."""
+    version = {}
+    for elem in ["table_name", "version_number", "last_modified", "institution", "contact"]:
+        version[elem] = found_or_none(tree.find(elem))
+    if version["table_name"] is None:
+        if (version["institution"] == "Centre for Environmental Data Analysis"
+                and version["contact"] == "support@ceda.ac.uk"):
+            version["table_name"] = "CF-StdNameTable"
+        else:
+            version["table_name"] = "USER-StdNameTable"
+    prettydict(outfile, "VERSION", version)
+    version_string = "-".join(version[k] for k in ["table_name", "version_number"])
+    outfile.write(f'CONVENTIONS_STRING = "{version_string}"\n\n')
+
+
+def write_useful_variables(outfile):
+    outfile.write(
+        '\n# The following variables are used for processing the standard names information below\n'
+        '_ACCEPT = "accept"\n'
+        '_WARN = "warn"\n'
+        '_REPLACE ="replace"\n'
+        '_ALTERNATIVE_MODES = [_ACCEPT, _WARN, _REPLACE]\n'
+        '_DEFAULT = "warn"\n'
+        '_MODE = _DEFAULT\n\n'
```

Comment:
We can see what you are trying to achieve here, but we would like to achieve it using Iris' existing structure instead:

### [`FUTURE` flag](https://github.com/SciTools/iris/blob/8604bb80977cb92bfe87d2486315311094219a31/lib/iris/__init__.py#L141)

Replacement is introduced as an opt-in setting. In future - typically the next major release - it will switch to being opt-out.

### Always warn if replacing

Since you made the proposal, we have agreed a global approach to Iris warnings: [Filtering Warnings — Iris 3.9.0 documentation (scitools-iris.readthedocs.io)](https://scitools-iris.readthedocs.io/en/stable/further_topics/filtering_warnings.html). The idea is that Iris should warn for EVERYTHING that might be a problem for a user, and the user applies warning filters if they don't wish to see a specific type of warning.